sycl/test/basic_tests/macros_no_rdc.cpp

```diff
@@ -0,0 +1,22 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple -fsycl-device-only -E -dD -fno-sycl-rdc %s -o %t.device
+// RUN: %clangxx %fsycl-host-only -fno-sycl-rdc -E -dD %s -o %t.host
+//
+// RUN: FileCheck --match-full-lines %s < %t.device --check-prefixes=DEVICE-FULL-LINE
+// RUN: FileCheck --match-full-lines %s < %t.host --check-prefixes=HOST
+//
+// Remove __DPCPP_SYCL_EXTERNAL to simplify regex for DEVICE prefix
+// RUN: sed -i 's|__DPCPP_SYCL_EXTERNAL||g' %t.device
```

Comment:
Do we have `sed` available on Windows? Or is it a part of `llvm-lit` built-in tools?

Reply:
I just grepped clang/test for sed and there are a ton of uses, and none of them seem to mark Windows unsupported.

and it's listed in the required tools section here:

![image](https://user-images.githubusercontent.com/7064844/223793408-28e93343-67e9-4d7a-a5c3-c89543d38048.png)



so I think it's fine

Reply:
We can simplify the test and avoid using tools like sed by using different name for DPCPP specific macro (e.g. __DPCPP_DEVICE).

Reply:
we can do that, but i don't think we should do that just for the test, we should do it if we think it's a better macro name.

Reply:
> I just grepped clang/test for sed and there are a ton of uses, and none of them seem to mark Windows unsupported.

Yeah, this seems like a good enough proof to me. I suggest we don't block the merge by this comment